From: Alexander Mikhaylenko <alexm@gnome.org>
Date: Wed, 24 Feb 2021 18:57:28 +0500
Subject: header-bar: Hide keyboard shortcuts button on mobile

GtkShortcutsWindow is not adaptive and not very useful on mobile to begin
with. Just hide it altogether.
---
 src/ephy-header-bar.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/ephy-header-bar.c b/src/ephy-header-bar.c
index 057f4c6..9af2d39 100644
--- a/src/ephy-header-bar.c
+++ b/src/ephy-header-bar.c
@@ -68,6 +68,8 @@ struct _EphyHeaderBar {
   GtkWidget *page_menu_popover;
 
   guint popover_hide_timeout_id;
+
+  GtkWidget *shortcuts_button;
 };
 
 G_DEFINE_TYPE (EphyHeaderBar, ephy_header_bar, GTK_TYPE_HEADER_BAR)
@@ -313,6 +315,9 @@ ephy_header_bar_constructed (GObject *object)
     gtk_widget_destroy (GTK_WIDGET (gtk_builder_get_object (builder, "run-in-background-button")));
   }
 
+  if (ephy_embed_shell_get_mode (embed_shell) != EPHY_EMBED_SHELL_MODE_APPLICATION)
+    header_bar->shortcuts_button = GTK_WIDGET (gtk_builder_get_object (builder, "keyboard-shortcuts-button"));
+
   header_bar->combined_stop_reload_button = GTK_WIDGET (gtk_builder_get_object (builder, "combined_stop_reload_button"));
   header_bar->combined_stop_reload_image = GTK_WIDGET (gtk_builder_get_object (builder, "combined_stop_reload_image"));
   gtk_widget_set_tooltip_text (header_bar->combined_stop_reload_button, _(REFRESH_BUTTON_TOOLTIP));
@@ -447,12 +452,16 @@ ephy_header_bar_set_adaptive_mode (EphyHeaderBar    *header_bar,
       gtk_revealer_set_reveal_child (GTK_REVEALER (header_bar->start_revealer), TRUE);
       gtk_revealer_set_reveal_child (GTK_REVEALER (header_bar->end_revealer), TRUE);
       gtk_widget_set_visible (header_bar->combined_stop_reload_button, FALSE);
+      if (header_bar->shortcuts_button)
+        gtk_widget_set_visible (header_bar->shortcuts_button, TRUE);
 
       break;
     case EPHY_ADAPTIVE_MODE_NARROW:
       gtk_revealer_set_reveal_child (GTK_REVEALER (header_bar->start_revealer), FALSE);
       gtk_revealer_set_reveal_child (GTK_REVEALER (header_bar->end_revealer), FALSE);
       gtk_widget_set_visible (header_bar->combined_stop_reload_button, TRUE);
+      if (header_bar->shortcuts_button)
+        gtk_widget_set_visible (header_bar->shortcuts_button, FALSE);
 
       break;
   }
